src/port/port.tcl

```diff
@@ -5693,6 +5693,29 @@ if {[catch {mportinit ui_options global_options global_variations} result]} {
     fatal "Failed to initialize MacPorts, $result"
 }
 
+# Change to port directory if requested
+if {[info exists global_options(ports_dir)]} {
+    set dir $global_options(ports_dir)
+    if {[string first "/" $dir] == -1} {
```

Comment:
Isn't this a change of behavior when specifying relative paths to port directories, e.g. when I'm in `/opt/myportstree/devel` and I specify `port -D foobar`, I may end up in `/opt/dports/textproc/foobar` if that's where portindex thinks the `foobar` port is.

This may not be a big issue in practice, but I'd like to make sure we're aware of it before merging this.

Reply:
I don't think this will be an issue. You can always force to use the directory with -D ./foobar` or `-D foobar/`.

Reply:
I agree, but it is a change of behavior. Anyway, `-D` isn't used very often, so I don't think that's an issue in this case.